tests/integration/api_container_test.py

```diff
@@ -620,6 +620,32 @@ def test_create_with_volume_mount(self):
         assert mount['Source'] == mount_data['Name']
         assert mount_data['RW'] is True
 
+    @requires_api_version('1.45')
+    def test_create_with_subpath_volume_mount(self):
+        mount = docker.types.Mount(
+            type="volume", source=helpers.random_name(),
+            target=self.mount_dest, read_only=True,
+            subpath='subdir'
+        )
+        host_config = self.client.create_host_config(mounts=[mount])
+        container = self.client.create_container(
+            TEST_IMG, ['true'], host_config=host_config,
+        )
+        assert container
+        inspect_data = self.client.inspect_container(container)
+        assert 'Mounts' in inspect_data
+        filtered = list(filter(
+            lambda x: x['Destination'] == self.mount_dest,
+            inspect_data['Mounts']
+        ))
+
+        print(mount)
+        assert len(filtered) == 1
+        mount_data = filtered[0]
+        assert mount['Source'] == mount_data['Name']
+        assert mount_data['RW'] is False
+        assert mount["VolumeOptions"]["Subpath"] == "subdir"
```

Comment:
This is testing that the input mount options, as defined above on line 628, are the same. 

The test needs to check that the path mounted _inside the container_ corresponds to the subdir as defined in the mount options. The volume mounted must therefore contain the `subdir`, or else there should be an error.

Maybe you could change to check `mount_data` instead, if it contains the necessary info? I don't have much time now but I'd have to check how these volume tests are defined in a bit more detail later

Reply:
Oh, sure